nix/status-go/default.nix

```diff
@@ -0,0 +1,123 @@
+{ stdenv, buildGoPackage, go, pkgs, fetchFromGitHub, androidPkgs, openjdk, xcodeWrapper }:
+
+with stdenv;
+
+let
+   gomobile = pkgs.callPackage ./gomobile { inherit androidPkgs xcodeWrapper; };
+   version = lib.fileContents ../../STATUS_GO_VERSION; # TODO: Simplify this path search with lib.locateDominatingFile
+   owner = "status-im";
+   repo = "status-go";
+   goPackagePath = "github.com/${owner}/${repo}";
+   rev = version;
+   sha256 = lib.fileContents ../../STATUS_GO_SHA256;
+   mobileTarget = if isDarwin then "ios" else "android";
+   mobileOutputFileName = if isDarwin then "Statusgo.framework" else "status-go-${version}.aar";
+   desktopOutputFileName = "libstatus.a";
+   destopSystem = hostPlatform.system;
+   removeReferences = [ go ];
+   removeExpr = refs: ''remove-references-to ${lib.concatMapStrings (ref: " -t ${ref}") refs}'';
+   goBuildFlags = "-v";
+   goBuildLdFlags = "-ldflags=-s";
+
+in buildGoPackage rec {
+  inherit goPackagePath version rev;
+  name = "${repo}-${version}";
+
+  src = pkgs.fetchFromGitHub { inherit rev owner repo sha256; };
+
+  nativeBuildInputs = [ gomobile openjdk ]
+    ++ lib.optional isDarwin xcodeWrapper;
+
+  # Fixes Cgo related build failures (see https://github.com/NixOS/nixpkgs/issues/25959 )
+  hardeningDisable = [ "fortify" ];
+
+  # gomobile doesn't seem to be able to pass -ldflags with multiple values correctly to go build, so we just patch files here  
+  patchPhase = ''
+    date=$(date -u '+%Y-%m-%d.%H:%M:%S')
+
+    substituteInPlace cmd/statusd/main.go --replace \
+      "buildStamp = \"N/A\"" \
+      "buildStamp = \"$date\""
+    substituteInPlace params/version.go --replace \
+      "var Version string" \
+      "var Version string = \"${version}\""
+    substituteInPlace params/version.go --replace \
+      "var GitCommit string" \
+      "var GitCommit string = \"${rev}\""
+    substituteInPlace vendor/github.com/ethereum/go-ethereum/metrics/metrics.go --replace \
+      "var EnabledStr = \"false\"" \
+      "var EnabledStr = \"true\""
+  '';
+
+  # we print out the version so that we fail fast in case there's any problem running xcrun, instead of failing at the end of the build
+  preConfigure = lib.optionalString isDarwin ''
+    xcrun xcodebuild -version
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    runHook renameImports
+
+    pushd "$NIX_BUILD_TOP/go/src/${goPackagePath}" >/dev/null
+
+    # TODO: Build desktop libraries
+    echo
+    echo "Building desktop libraries"
```

Comment:
why do we print that we are "Building desktop libraries" when we don't? that will be confusing to debug

Reply:
huh? We *are* building them (just not the Windows ones). I do need to remove the TODO.

Reply:
then this TODO is confusing 😛